lib/internal/modules/cjs/loader.js

```diff
@@ -584,34 +584,28 @@ function resolveExportsTarget(pkgPath, target, subpath, basePath, mappingKey) {
       }
     }
   } else if (typeof target === 'object' && target !== null) {
-    if (experimentalConditionalExports &&
-        ObjectPrototypeHasOwnProperty(target, 'node')) {
-      try {
-        const result = resolveExportsTarget(pkgPath, target.node, subpath,
-                                            basePath, mappingKey);
-        emitExperimentalWarning('Conditional exports');
-        return result;
-      } catch (e) {
-        if (e.code !== 'MODULE_NOT_FOUND') throw e;
-      }
-    }
-    if (experimentalConditionalExports &&
-        ObjectPrototypeHasOwnProperty(target, 'require')) {
-      try {
-        const result = resolveExportsTarget(pkgPath, target.require, subpath,
-                                            basePath, mappingKey);
-        emitExperimentalWarning('Conditional exports');
-        return result;
-      } catch (e) {
-        if (e.code !== 'MODULE_NOT_FOUND') throw e;
-      }
-    }
-    if (ObjectPrototypeHasOwnProperty(target, 'default')) {
-      try {
-        return resolveExportsTarget(pkgPath, target.default, subpath,
-                                    basePath, mappingKey);
-      } catch (e) {
-        if (e.code !== 'MODULE_NOT_FOUND') throw e;
+    for (const p of ObjectKeys(target)) {
+      switch (p) {
+        case 'node':
+        case 'require':
+          if (!experimentalConditionalExports)
+            continue;
+          try {
+            const result = resolveExportsTarget(pkgPath, target[p], subpath,
+                                                basePath, mappingKey);
+            emitExperimentalWarning('Conditional exports');
```

Comment:
should this be above, when we actually call resolveExportsTarget so that if it throws it still prints the warning?

Reply:
Good point.